From b48ac499ac06f0ef757bdd3a3eb4f8886eb12119 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 220/296] set the resultTo to null started by a finishing
 activity

activitya start activityb via startActivityForResult
if activitya is going to finish, activityb's resultTo
should be set null. otherwise, activityb will be put
to the topTask which is not supposed to happen instead of
creating a new taskrecord.
---
 services/core/java/com/android/server/am/ActivityStarter.java | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/services/core/java/com/android/server/am/ActivityStarter.java b/services/core/java/com/android/server/am/ActivityStarter.java
index 7575f71e90b..435c97b79e2 100644
--- a/services/core/java/com/android/server/am/ActivityStarter.java
+++ b/services/core/java/com/android/server/am/ActivityStarter.java
@@ -1481,6 +1481,9 @@ class ActivityStarter {
         }
         mSourceRecord = null;
         mSourceStack = null;
+        if (mStartActivity != null) {
+            mStartActivity.resultTo = null;
+        }
     }
 
     /**
-- 
2.11.0

